test/index.test.js

```diff
@@ -67,48 +71,105 @@ describe('Index Tests', () => {
     });
   }
 
-  it('uploads images to media-bus', async () => {
+  describe('image upload', () => {
     const testImagePath = resolve(__testdir, 'fixtures', '300.png');
-    nock.fstab();
-    nock('https://www.example.com')
-      .get('/blog/article')
-      .replyWithFile(200, resolve(__testdir, 'fixtures', 'images.html'), {
-        'last-modified': 'Sat, 22 Feb 2031 15:28:00 GMT',
-      })
-      .get('/absolute.png')
-      .basicAuth({ user: 'john', pass: 'doe' })
-      .replyWithFile(200, testImagePath, {
-        'content-type': 'image/png',
-      })
-      .get('/blog/relative.png')
-      .replyWithFile(200, testImagePath, {
-        'content-type': 'image/png',
-      })
-      .get('/missing.png')
-      .reply(404);
-    nock('https://dummyimage.com')
-      .get('/300.png')
-      .replyWithFile(200, testImagePath, {
-        'content-type': 'image/png',
+
+    before(() => {
+      // mock the image and related s3 requests for existing images only once as they will be
+      // cached by the MediaHandler
```

Comment:
I'd prefer to add an option to disable the cache instead of relying on side effects in the tests.